### PR TITLE
Fix pseudo-selector order

### PR DESCRIPTION
**Pull request by itamargiv**
This PR aims to fix an issue with hover styles by correcting the pseudo selector order to reflect the overrides performed by the user. i.e. an element is:

1. First `:focus`ed (most general case)
2. Then `:hover`ed (specific case before activated, desktop only)
3. And only then potentially `:active`ated

In this case the selector cascade reflects that flow and thus does not require a higher specificity to perform overrides.

See: https://phabricator.wikimedia.org/T266557
Bug: T266557

### CODE REVIEW

**Comment by SaiSan-WMDE** [comment]
Would it be possible to prevent the buttons' border from changing whenever a focused button is hovered? Only the button text color and the background color should be affected.

**Review by jakobw — CHANGES_REQUESTED** [review]
It bothers me a little bit to have `:focus:hover` mix and match styling from the two states without making this visible in the tokens. Was this considered? :thinking: 

Looks good in general and I like how the pseudo selectors are reordered! I left one small suggestion inline.

File: vue-components/src/components/Button.vue [hunk 1]
```diff
@@ -127,23 +127,27 @@ $base: '.wikit-Button';
...
 			}
-			// A clicked button is both :active and :focused. Using :not(:active) to avoid mixing the two.
-			&:focus:not(:active) {
-				color: $wikit-Button-normal-neutral-focus-color;
-				background-color: $wikit-Button-normal-neutral-focus-background-color;
-				border-color: $wikit-Button-normal-neutral-focus-border-color;
-				box-shadow: $wikit-Button-normal-neutral-focus-box-shadow;
+
+			&:hover:not(:focus) {
+				border-color: $wikit-Button-normal-neutral-hover-border-color;
```
[the remark this answers is not in the document]
Comment (SaiSan-WMDE) [reply vue-components/src/components/Button.vue:150]: Hey Jakob. So you mean that we should consider creating `$wikit-Button-xx-xx-focus-hovered` tokens? I thought about it, but concluded that creating new styles with the same values sounded a bit redundant :-/ Representing one pseudo selector level in the tokens seems enough. After all, the combination of pseudo selectors (in this or other cases) could be considered an implementation decision, right?

**Comment by jakobw** [comment]
(I'm answering here because I don't think the comment thread is related to the code suggestion.)

Hey @SaiSan-WMDE!

> So you mean that we should consider creating $wikit-Button-xx-xx-focus-hovered tokens?

Something like that, yup!

> Representing one pseudo selector level in the tokens seems enough. After all, in general the combination of pseudo selectors (in this or other cases) could be considered an implementation decision, right?

I would agree if there was some cascading order that would lead to the desired behavior. Overriding some, but not all of the `:focus` properties on `:hover` means that there is no way we could arrive at the expected styling for the combination of the two based on the existing tokens.

It seems that this was considered though, so I'm ok with the decision! The change request was only for the code suggestion and not about the tokens.

**Comment by SaiSan-WMDE** [comment]
> (I'm answering here because I don't think the comment thread is related to the code suggestion.)

Sorry about the wrongly nested response.

> I would agree if there was some cascading order that would lead to the desired behavior. Overriding some, but not all of the `:focus` properties on `:hover` means that there is no way we could arrive at the expected styling for the combination of the two based on the existing tokens.

I see the point and I agree in this instance, but I'd tentatively move forward expressing only single states styles in the tokens to avoid introducing redundancy.

Thanks so much for your feedback, Jakob 💯

**Review by jakobw — DISMISSED** [review]
🚀

**Review by SaiSan-WMDE — APPROVED** [review]

**Pull request merged**